tests/st/cycle_gan.py

```diff
@@ -0,0 +1,252 @@
+# Copyright 2020-2021 Huawei Technologies Co., Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""Cycle GAN Tutorial
+The sample can be run on CPU, GPU and Ascend 910 AI processor.
+"""
+import os
+import argparse
+
+import tinyms as ts
+from tinyms import context, Tensor
+from tinyms.data import GeneratorDataset, UnalignedDataset, GanImageFolderDataset, DistributedSampler
+from tinyms.vision import cyclegan_transform
+from tinyms.model.cycle_gan.cycle_gan import get_generator_discriminator, cycle_gan, TrainOneStepG, TrainOneStepD
+from tinyms.losses import DiscriminatorLoss, GeneratorLoss
+from tinyms.optimizers import Adam
+from tinyms.data.utils import save_image, generate_image_list
+from tinyms.utils.common_utils import GanReporter, gan_load_ckpt, GanImagePool
+from tinyms.utils.train import cyclegan_lr
+from tinyms.utils.eval import CityScapes, fast_hist, get_scores
+
+
+def create_dataset(dataset_path, batch_size=1, repeat_size=1, max_dataset_size=None,
+                   shuffle=True, num_parallel_workers=1, phase='train', data_dir='testA'):
+    """ create Mnist dataset for train or eval.
+    Args:
+        data_path: Data path
+        batch_size: The number of data records in each group
+        repeat_size: The number of replicated data records
+        num_parallel_workers: The number of parallel workers
+    """
+    # define dataset and apply the transform func
+    if phase == 'train':
+        ds = UnalignedDataset(dataset_path, phase, max_dataset_size=max_dataset_size, shuffle=True)
+
+        device_num = 1
+        distributed_sampler = DistributedSampler(len(ds), num_replicas=device_num, rank=0, shuffle=shuffle)
+        gan_generator_ds = GeneratorDataset(ds, column_names=["image_A", "image_B"], sampler=distributed_sampler,
+                                            num_parallel_workers=num_parallel_workers)
+    else:
+        datadir = os.path.join(dataset_path, data_dir)
+        ds = GanImageFolderDataset(datadir, max_dataset_size=max_dataset_size)
+        gan_generator_ds = GeneratorDataset(ds, column_names=["image", "image_name"],
+                                            num_parallel_workers=num_parallel_workers)
+
+    gan_generator_ds = cyclegan_transform.apply_ds(gan_generator_ds,
```

Comment:
Could u set the `phase` argument to `is_training` to keep consistent with other dataset transforms?